### DISCUSSION TITLE
Problem when installing symfony webapp

### DISCUSSION THREAD

**Discussion by cosminsandu** [symfony/symfony#54145] [Q&A]
### Symfony version(s) affected

6.4.*

### Description

When I try to install symfony 
> symfony new my_project_directory --version="6.4.*" --webapp

i get the following error :

Script importmap:require returned with error code 1
!!
!!  In ErrorChunk.php line 56:
!!
!!    SSL certificate problem: certificate has expired for "https://data.jsdelivr.com/v1/packages/npm/@hotwired/stimulus/resolved?specifier=%5E3.0.0".
!!
!!
!!  In CurlResponse.php line 315:
!!
!!    SSL certificate problem: certificate has expired for "https://data.jsdelivr.com/v1/packages/npm/@hotwired/stimulus/resolved?specifier=%5E3.0.0".
!!
![image](https://github.com/symfony/symfony/assets/3448525/fa4b57ab-d6f0-4626-94c1-1793f5be7eb8)

### How to reproduce

just run:
> symfony new my_project_directory --version="6.4.*" --webapp
or 
> composer create-project symfony/skeleton:"6.4.*" my_project_directory
> cd my_project_directory
> composer require webapp

### Additional Context

OS: Windows 11 Pro 23H2 build 22631.3155

PHP 8.1.7 (cli) (built: Jun  7 2022 21:45:03) (NTS Visual C++ 2019 x64)
Copyright (c) The PHP Group
Zend Engine v4.1.7, Copyright (c) Zend Technologies
    with Xdebug v3.1.5, Copyright (c) 2002-2022, by Derick Rethans

Composer version 2.4.1 2022-08-20 11:44:50

npm -v
6.14.16

**Comment by derrabus** [comment]
I can't reproduce this issue. Moving to support.

**Reply by cosminsandu** [reply]
On WSL it works ok.

**Comment by cosminsandu** [comment]
Here is a video displaying the problem:
https://www.loom.com/share/9411d470b53743aeb076714ce01d23e2?sid=97d38e5c-8abe-44ef-b755-75570bf0f9fb

**Comment by cosminsandu** [comment]
@weaverryan 

This seem to be related to @hotwired/stimulus/

do you have any idea what is going on?

**Comment by cosminsandu** [comment]
The problem persists!

I've updated the Context:

OS: Windows 11 Pro 23H2 build 22631.3155

PHP 8.1.7 (cli) (built: Jun 7 2022 21:45:03) (NTS Visual C++ 2019 x64)
Copyright (c) The PHP Group
Zend Engine v4.1.7, Copyright (c) Zend Technologies
with Xdebug v3.1.5, Copyright (c) 2002-2022, by Derick Rethans

Composer version 2.4.1 2022-08-20 11:44:50

Please let me know if any other info are nedeed.

**Comment by cosminsandu** [comment]
The error message was updated:
>  SSL certificate problem: *self signed certificate in certificate chain* for "https://data.jsdelivr.com/v1/packages/npm/@hotwired/stimulus/resolved?specifier=%5E3.0.0".

![image](https://github.com/symfony/symfony/assets/3448525/d93e0473-600b-4f45-a344-380055ac5383)

**Reply by cosminsandu** [reply]
This is related to _importmap:require_ command. 

Then running this command in verbose mode I get:

> d:\WORK\my_project_directory>php bin/console importmap:require lodash -vvv
> 13:51:55 INFO      [http_client] Request: "GET https://data.jsdelivr.com/v1/packages/npm/lodash/resolved"
> 
> In ErrorChunk.php line 56:
> 
>   [Symfony\Component\HttpClient\Exception\TransportException]
>   SSL certificate problem: self signed certificate in certificate chain for "https://data.jsdelivr.com/v1/packages/npm/lodash/resolved".
> 
> 
> Exception trace:
>   at D:\WORK\my_project_directory\vendor\symfony\http-client\Chunk\ErrorChunk.php:56
>  Symfony\Component\HttpClient\Chunk\ErrorChunk->isFirst() at D:\WORK\my_project_directory\vendor\symfony\http-client\Response\CommonResponseTrait.php:149
>  Symfony\Component\HttpClient\Response\CurlResponse::initialize() at D:\WORK\my_project_directory\vendor\symfony\http-client\Response\TransportResponseTrait.php:52
>  Symfony\Component\HttpClient\Response\CurlResponse->getStatusCode() at D:\WORK\my_project_directory\vendor\symfony\http-client\Response\TraceableResponse.php:73
>  Symfony\Component\HttpClient\Response\TraceableResponse->getStatusCode() at D:\WORK\my_project_directory\vendor\symfony\asset-mapper\ImportMap\Resolver\JsDelivrEsmResolver.php:73
>  Symfony\Component\AssetMapper\ImportMap\Resolver\JsDelivrEsmResolver->resolvePackages() at D:\WORK\my_project_directory\vendor\symfony\asset-mapper\ImportMap\ImportMapManager.php:176
>  Symfony\Component\AssetMapper\ImportMap\ImportMapManager->requirePackages() at D:\WORK\my_project_directory\vendor\symfony\asset-mapper\ImportMap\ImportMapManager.php:123
>  Symfony\Component\AssetMapper\ImportMap\ImportMapManager->updateImportMapConfig() at D:\WORK\my_project_directory\vendor\symfony\asset-mapper\ImportMap\ImportMapManager.php:43
>  Symfony\Component\AssetMapper\ImportMap\ImportMapManager->require() at D:\WORK\my_project_directory\vendor\symfony\asset-mapper\Command\ImportMapRequireCommand.php:113
>  Symfony\Component\AssetMapper\Command\ImportMapRequireCommand->execute() at D:\WORK\my_project_directory\vendor\symfony\console\Command\Command.php:326
>  Symfony\Component\Console\Command\Command->run() at D:\WORK\my_project_directory\vendor\symfony\console\Application.php:1096
>  Symfony\Component\Console\Application->doRunCommand() at D:\WORK\my_project_directory\vendor\symfony\framework-bundle\Console\Application.php:126
>  Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at D:\WORK\my_project_directory\vendor\symfony\console\Application.php:324
>  Symfony\Component\Console\Application->doRun() at D:\WORK\my_project_directory\vendor\symfony\framework-bundle\Console\Application.php:80
>  Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at D:\WORK\my_project_directory\vendor\symfony\console\Application.php:175
>  Symfony\Component\Console\Application->run() at D:\WORK\my_project_directory\vendor\symfony\runtime\Runner\Symfony\ConsoleApplicationRunner.php:49
>  Symfony\Component\Runtime\Runner\Symfony\ConsoleApplicationRunner->run() at D:\WORK\my_project_directory\vendor\autoload_runtime.php:29
>  require_once() at D:\WORK\my_project_directory\bin\console:11
> 
> In CurlResponse.php line 315:
> 
>   [Symfony\Component\HttpClient\Exception\TransportException]
>   SSL certificate problem: self signed certificate in certificate chain for "https://data.jsdelivr.com/v1/packages/npm/lodash/resolved".
> 
> 
> Exception trace:
>   at D:\WORK\my_project_directory\vendor\symfony\http-client\Response\CurlResponse.php:315
>  Symfony\Component\HttpClient\Response\CurlResponse::perform() at D:\WORK\my_project_directory\vendor\symfony\http-client\Response\TransportResponseTrait.php:167
>  Symfony\Component\HttpClient\Response\CurlResponse::stream() at D:\WORK\my_project_directory\vendor\symfony\http-client\Response\CommonResponseTrait.php:148
>  Symfony\Component\HttpClient\Response\CurlResponse::initialize() at D:\WORK\my_project_directory\vendor\symfony\http-client\Response\TransportResponseTrait.php:52
>  Symfony\Component\HttpClient\Response\CurlResponse->getStatusCode() at D:\WORK\my_project_directory\vendor\symfony\http-client\Response\TraceableResponse.php:73
>  Symfony\Component\HttpClient\Response\TraceableResponse->getStatusCode() at D:\WORK\my_project_directory\vendor\symfony\asset-mapper\ImportMap\Resolver\JsDelivrEsmResolver.php:73
>  Symfony\Component\AssetMapper\ImportMap\Resolver\JsDelivrEsmResolver->resolvePackages() at D:\WORK\my_project_directory\vendor\symfony\asset-mapper\ImportMap\ImportMapManager.php:176
>  Symfony\Component\AssetMapper\ImportMap\ImportMapManager->requirePackages() at D:\WORK\my_project_directory\vendor\symfony\asset-mapper\ImportMap\ImportMapManager.php:123
>  Symfony\Component\AssetMapper\ImportMap\ImportMapManager->updateImportMapConfig() at D:\WORK\my_project_directory\vendor\symfony\asset-mapper\ImportMap\ImportMapManager.php:43
>  Symfony\Component\AssetMapper\ImportMap\ImportMapManager->require() at D:\WORK\my_project_directory\vendor\symfony\asset-mapper\Command\ImportMapRequireCommand.php:113
>  Symfony\Component\AssetMapper\Command\ImportMapRequireCommand->execute() at D:\WORK\my_project_directory\vendor\symfony\console\Command\Command.php:326
>  Symfony\Component\Console\Command\Command->run() at D:\WORK\my_project_directory\vendor\symfony\console\Application.php:1096
>  Symfony\Component\Console\Application->doRunCommand() at D:\WORK\my_project_directory\vendor\symfony\framework-bundle\Console\Application.php:126
>  Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at D:\WORK\my_project_directory\vendor\symfony\console\Application.php:324
>  Symfony\Component\Console\Application->doRun() at D:\WORK\my_project_directory\vendor\symfony\framework-bundle\Console\Application.php:80
>  Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at D:\WORK\my_project_directory\vendor\symfony\console\Application.php:175
>  Symfony\Component\Console\Application->run() at D:\WORK\my_project_directory\vendor\symfony\runtime\Runner\Symfony\ConsoleApplicationRunner.php:49
>  Symfony\Component\Runtime\Runner\Symfony\ConsoleApplicationRunner->run() at D:\WORK\my_project_directory\vendor\autoload_runtime.php:29
>  require_once() at D:\WORK\my_project_directory\bin\console:11
> 
> importmap:require [--entrypoint] [--path PATH] [--] <packages>...


![image](https://github.com/symfony/symfony/assets/3448525/a8dc3169-4ef6-4ed9-aa00-69a08f9d1a72)

**Reply by stof** [reply]
Make sure you don't have an antivirus software performing MitM on your HTTPS requests. This is the most likely reason why data.jsdelivr.com might have a self-signed certificate instead of a trusted certificate.

**Reply by cosminsandu** [reply]
Hi @stof.

I've disabled my antivirus (Bitdefender total security) but the problem persists!

any other ideeas?

**Reply by simplegr** [reply]
I have the same problem on windows 10. I 've tried PHP 8.1.9 and 8.2.16.

**Reply by cosminsandu** [reply]
@simplegr do you have any antivirus?

**Reply by simplegr** [reply]
@cosminsandu I disabled the antivirus, still the same.

**Reply by cosminsandu** [reply]
Same for me - but I want to see if it's only on my side.

I managed to bypass this issue by working on WSL ....

**Comment by cosminsandu** [comment]
The problem persist on:
Microsoft Windows 11 Pro (Version	10.0.22631 Build 22631)
PHP 8.2.14 (cli) (built: Dec 20 2023 10:19:26) (NTS Visual C++ 2019 x64)
Composer version 2.4.1 2022-08-20 11:44:50

I get the same error when running: ``symfony new my_project_directory71 --version="7.1.*" --webapp``:
![image](https://github.com/user-attachments/assets/8fcbb49d-1c5d-459b-80fe-6b127ca88fe7)

**Reply by christian-fries** [reply]
For us, the problem was SSL inspection on network level. You can adjust the configuration of the [http_client](https://symfony.com/doc/current/reference/configuration/framework.html#http-client) to fix the problem. Two options:
1. Provide the path to the custom certificate as [cafile](https://symfony.com/doc/current/reference/configuration/framework.html#cafile) or [capath](https://symfony.com/doc/current/reference/configuration/framework.html#capath) (recommended)
2. Disable [verify_host](https://symfony.com/doc/current/reference/configuration/framework.html#verify-host) and [verify_peer](https://symfony.com/doc/current/reference/configuration/framework.html#verify-peer) (not recommended)

You can use `when@dev` to only apply the configuration in dev context.

**Comment by musaffarpatel** [comment]
I was able to solve this on my local windows / apache setup as follows :

Download the certificate from: https://curl.se/ca/cacert.pem
Save it locally, eg c:\certs\cacert.pem

In php.ini file, change or add the lines:

curl.cainfo = "c:\certs\cacert.pem"
and
openssl.cafile = "c:\certs\cacert.pem"
